components/CreateClassModal/CreateClassModal.js

```diff
@@ -0,0 +1,169 @@
+                <DateTimePicker
+                  required
+                  renderInput={(params) => <TextField {...params} fullWidth />}
+                  label="New Start Time"
+                  value={newStartTime}
+                  name="start_date"
+                  onChange={(newTimeValue) => {
+                    setClassObj({ ...classObj, 'startTime': newTimeValue })
+                    setNewStartTime(newTimeValue)
+                  }}
+                />
+              </LocalizationProvider>
+               {/* ----------------------END TIME PICKER-------------------------- */}
+              <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <DateTimePicker
+                  required
+                  renderInput={(props) => <TextField {...props} fullWidth />}
+                  label="New End Time"
+                  name="end_date"
+                  value={newEndTime}
+                  onChange={(newTimeValue) => {
+                    setClassObj({ ...classObj, 'endTime': newTimeValue });
+                    setNewEndTime(newTimeValue)
+                  }}
+                />
+              </LocalizationProvider>
+            </Stack>
+            {/* ----------------------ZOOM LINK-------------------------- */}
+            <TextField
+              required
+              className={styles.classForm}
+              name="meeting_url"
+              margin="dense"
+              label="Classroom Link"
+              type="text"
+              color='primary'
+              onChange={handleChange}
+              placeholder="Please enter your Zoom Meeting Room Link"
+            />
+            {/* ----------------------TYPE SELECTOR-------------------------- */}
+            <FormControl className={styles.classForm}>
+              <InputLabel>Privacy Type Settings</InputLabel>
+              <Select
+                required
+                label="Privacy Type Settings"
+                defaultValue="Privacy"
+                value={type}
+                name="type"
+                margin="dense"
+                onChange={(e) => {
+                  handleChange(e);
+                  setType(e.target.value);
+                }}
+              >
+                <MenuItem value="Public">Public</MenuItem>
+                <MenuItem value="Private">Private</MenuItem>
+              </Select>
+            </FormControl>
+             {/* ----------------------CLASS CAPACITY-------------------------- */}
+            <TextField
+              required
+              className={styles.classSizeMargin}
+              name="capacity"
+              margin="dense"
+              label="Class Size"
+              type="Number"
+              color='primary'
+              onChange={handleChange}
+            />
+            <MainButton value="Submit" />
+          </FormControl>
+        </DialogContent>
+      </Dialog>
+    </div>
```

Comment:
Biggest 2 things in this file is some hard coded spots and I think useReducer hook could do wonders here. It works really well with form data.